packages/react-core/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Icon/icon';
+import { css } from '@patternfly/react-styles';
+import { Spinner } from '../Spinner';
+
+export interface IconComponentProps extends Omit<React.HTMLProps<HTMLSpanElement>, 'size'> {
+  /** Icon content when isInProgress is set to false */
+  children?: React.ReactNode;
+  /** Icon content when isInProgress is set to true. Defaults to a medium spinner with 1em diameter.  */
+  progressIcon?: React.ReactNode;
+  /** Additional classes applied to the icon container */
+  className?: string;
+  /** Size of the icon */
+  size?: 'sm' | 'md' | 'lg' | 'xl';
+  /** Size of icon content. Overrides size property. */
+  iconSize?: 'sm' | 'md' | 'lg' | 'xl';
+  /** Status color of the icon */
+  status?: 'default' | 'info' | 'success' | 'warning' | 'danger';
+  /** Indicates the icon is inline and should inherit the text font size and color. Overriden by size and iconSize properties. */
+  isInline?: boolean;
+  /** Indicates the icon is in progress. Setting this property to true will swap the icon content to the progressIcon. */
+  isInProgress?: boolean;
+  /** Aria-label for the default progress icon */
+  defaultProgressArialabel?: string;
+}
+
+export const Icon: React.FunctionComponent<IconComponentProps> = ({
+  children,
+  className,
+  progressIcon,
+  size,
+  iconSize,
+  status,
+  isInline = false,
+  isInProgress = false,
+  defaultProgressArialabel = 'Loading...',
+  ...props
+}: IconComponentProps) => {
+  const _progressIcon = progressIcon ? (
+    progressIcon
+  ) : (
+    <Spinner size="md" diameter="1em" isSVG aria-label={defaultProgressArialabel} />
```

Comment:
```suggestion
    <Spinner size="md" diameter="1em" isSVG aria-label={defaultProgressArialabel} />
```

using `diameter` will always squash whatever is passed to `size` since they both set the `--pf-c-spinner--diameter` var, but the `size` modifier classes set that var in the CSS, and `diameter` sets it as an inline `style="..."` attribute, and the inline style will always win.

Also I noticed the core examples set the `size` to "md", too, (and they don't set the inline style to `1em`) though I think what you've done here is correct, since if you set `size` or `iconSize`, using `1em` as the size will update the spinner. Using the `size` prop, it will always stay the same size.

Though the `iconProgress` element also takes size props so you can size the progress icon independently of the icon size - that might be useful with our spinner or custom spinners, depending on the size of the "graphical" portion of the icon used in relation to the of the graphical portion of the spinner. Those can be different sometimes, or appear different, even if they're technically the same sized SVG or icon font char or whatever it is.